open-telemetry/opentelemetry-js
# chore(semconv): Separate SemConv releases to allow different versioning

## Which problem is this PR solving?

Sometimes folks get confused about the version of this package, expecting it to match the version of semantic conventions being produced. This becomes especially problematic as the package version number is now very close to the semantic convention version number. By releasing separately we can make the versions match the expectation.

Fixes #4878 

## Short description of the changes

- add new workspace for semconv
- add separate changelog for semconv
- update release scripts and workflow to separate semconv
- update contributing guides to clarify semconv is separate
- update folder name from `opentelemetry-semantic-conventions` to `semantic-conventions`
- move `semantic-conventions` outside of `packages` directory
- update all paths and references to the old name

This looks like a really big change, but it's mostly renaming files and updating paths.

## Type of change


## How Has This Been Tested?

To test release notes script:
- replace `Unreleased` in the CHANGELOG to an actual version, like `1.1.1`. Then run the script as described in the comments (`node scripts/extract-latest-release-notes.js ./semantic-conventions/CHANGELOG.md`). There should be a .tmp/release-notes.md with the contents of the changelog for that version.

Also tested in separate repo to confirm successful separation of release PRs

## Checklist:
